src/freenas-installer/etc/install.sh

```diff
@@ -12,13 +12,85 @@ export TERM
 
 . /etc/avatar.conf
 
+
+# Constants for base 10 and base 2 units
+: ${kB:=$((1000))}      ${kiB:=$((1024))};       readonly kB kiB
+: ${MB:=$((1000 * kB))} ${MiB:=$((1024 * kiB))}; readonly MB MiB
+: ${GB:=$((1000 * MB))} ${GiB:=$((1024 * MiB))}; readonly GB GiB
+: ${TB:=$((1000 * GB))} ${TiB:=$((1024 * GiB))}; readonly TB TiB
+
 is_truenas()
 {
 
     test "$AVATAR_PROJECT" = "TrueNAS"
     return $?
 }
 
+# Constant media size threshold for allowing swap partitions.
+: ${MIN_SWAPSAFE_MEDIASIZE:=$((60 * GB))}; readonly MIN_SWAPSAFE_MEDIASIZE
+
+# Check if it is safe to create swap partitions on the given disks.
+#
+# The result can be forced by setting SWAP_IS_SAFE in the environment to either
+# "YES" or "NO".
+#
+# Sets SWAP_IS_SAFE to "YES" if
+#   we are on TrueNAS
+#   *or*
+#   every disk in $@ is >= ${MIN_SWAPSAFE_MEDIASIZE} and none is USB and user says ok
+# Otherwise sets SWAP_IS_SAFE to "NO".
+#
+# Use `is_swap_safe` to check the value of ${SWAP_IS_SAFE}.
+check_is_swap_safe()
+{
+    # We assume swap is safe on TrueNAS,
+    # and we try to use the existing value for ${SWAP_IS_SAFE} if already set.
+    if ! is_truenas && [ -z "${SWAP_IS_SAFE}" ] ; then
+	local _disk
+	# Check every disk in $@, aborting if an unsafe disk is found.
+	for _disk ; do
+	    if [ $(diskinfo "${_disk}" | cut -f 3) -lt ${MIN_SWAPSAFE_MEDIASIZE} ] ||
+		camcontrol identify "${_disk}" -v 2>&1 | grep -qF 'umass-sim' ; then
```

Comment:
Why `identify`?  It won't show you anything for SCSI disk.  I proposed `negotiate ... -v` to get info about the HBA.

Reply:
Oh indeed you did! I typed identify and saw the string "umass-sim" in the output so I failed to recognize the mixup. Fix coming right up.